apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -138,31 +135,24 @@ public static void UseGoogleTrace(this IApplicationBuilder app)
             var tracerFactory = new ManagedTracerFactory(projectId, consumer,
                 RateLimitingTraceOptionsFactory.Create(config), traceIdFactory);
 
+            services.AddScoped(CreateTraceHeaderContext);
```

Comment:
This is used in the middleware to create the managed tracer.  We could take that logic and move it here too.